bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy

```diff
@@ -1478,6 +1507,10 @@ class ApiController {
     return true
   }
 
+  def extractFilenameFromUrl(String url) {
+    return url.split('/')[-1]
+  }
```

Comment:
You are supposing all the URLs will follow the same pattern, what is not true.

Reply:
https://stackoverflow.com/a/17167743

Reply:
In case it's impossible to discover the filename using the URL, a generic name such as `pre-uploaded-presentation` could be used, with the extension populated afterwards